lib/matplotlib/colorbar.py

```diff
@@ -854,10 +857,22 @@ def get_ticks(self, minor=False):
             return self._long_axis().get_majorticklocs()
 
     @_api.delete_parameter("3.5", "update_ticks")
-    def set_ticklabels(self, ticklabels, update_ticks=True):
+    def set_ticklabels(self, ticklabels, *, minor=False, update_ticks=True,
+                       **kwargs):
         """
         Set tick labels.
 
+        .. admonition:: Discouraged
+
+            The use of this method is discouraged, because of the dependency
+            on tick positions. In most cases, you'll want to use
+            ``set_ticks(positions, labels)`` instead.
+
+            If you are using this method, you should always fix the tick
+            positions before, e.g. by using `.Axis.set_ticks` or by explicitly
+            setting a `~.ticker.FixedLocator`. Otherwise, ticks are free to
```

Comment:
Probably not `Axis.set_ticks`, but the colorbar method instead; and, is there a way to set a locator on the colorbar?

Reply:
Not explicitly, though you can pass down to the axis and do so.  Changed the wording.  However, as noted above the formatter and locator of the colorbar should probably just reflect exactly what is on the axis.